### ISSUE TITLE
Buffered read / writes for the EEPROM emulation

### ISSUE THREAD

**Issue by hasenbanck** [stm32duino/Arduino_Core_STM32#296]
Right now the EEPROM emulation inside the core (stm32_eeprom.c/h) provides two functions for reading and writing to an EEPROM emulated via flash (size of a page).

Both functions are touching the whole page in each function call, which is especially wasteful when writing to the flash.

To stay backward-compatible, I would propose to add four new functions to the core library:

```C
void eeprom_buffer_fill();
void eeprom_buffer_flush();
uint8_t eeprom_buffered_read_byte(const uint16_t __p);
void eeprom_buffered_write_byte(uint16_t __p, uint8_t __value);
```

The developer would then to first fill the static, private buffer with the fill function, interact with it and when he wants to save the content of the buffer to flash, flush the buffer to flash.

Implementing these changes shouldn't take much effort, since all pieces are already in place. I would like to do the implementation.

Comments? Suggestions?

**Comment by fpistm** [comment]
The EEPROM is not really optimized. 
This feature is mainly to provide easiest way to save/restore some user settings which don't required high data rate and access else a real storage (SD,...) should be used.
Anyway, your suggestions seems a good improvement.

**Comment by hasenbanck** [comment]
Since I'm currently adapting Marlin for your core, this problems popped up. Currently Marlin would safe the configuration (around 600 bytes) one byte at a time and would use 600 erase cycles of the whole page just for one save operation.

I implemented the usage of the Backup SRAM as an alternative, but I want to correct the current default operation mode non the less.

I will open a PR once I finished and tested the implementation.
